File: packages/go/contracts/commands/liverun.go
package commands

import "trend-buddy/packages/go/contracts/domain"

type LiveRunControlAction string

const (
	LiveRunControlStart LiveRunControlAction = "start"
	LiveRunControlPause LiveRunControlAction = "pause"
	LiveRunControlStop  LiveRunControlAction = "stop"
)

type LiveRunControlRequest struct {
	RunID        domain.LiveRunID     `json:"run_id,omitempty"`
	StrategyID   domain.StrategyID    `json:"strategy_id"`
	InstrumentID domain.InstrumentID  `json:"instrument_id"`
	Action       LiveRunControlAction `json:"action"`
	Parameters   map[string]any       `json:"parameters,omitempty"`
}

File: packages/python/src/trend_buddy_shared/contracts/commands/liverun.py
from dataclasses import dataclass, field
from enum import StrEnum
from typing import Any


class LiveRunControlAction(StrEnum):
    START = "start"
    PAUSE = "pause"
    STOP = "stop"


@dataclass
class LiveRunControlRequest:
    strategy_id: str
    instrument_id: str
    action: LiveRunControlAction
    run_id: str = ""
    parameters: dict[str, Any] = field(default_factory=dict)
